src/Language/PureScript/Ide/Types.hs

```diff
@@ -293,3 +295,10 @@ instance ToJSON PursuitResponse where
       , "package" .= package
       , "text"    .= text
       ]
+
+-- TODO: better names?
+data IdeDeclName
+  = IdeNameValue Text
+  | IdeNameType Text
+  | IdeNameKind Text
+  deriving (Show, Eq, Ord)
```

Comment:
@kRITZCREEK is this okay for replacing the use of `Either Text Text` that you mentioned?